### PR TITLE
bump fog openstack gem to new version 1.1.1 & enable application credential authentication

### PR DESCRIPTION
**Pull request by fmoehler**

### CODE REVIEW

File: src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb [hunk 1]
```diff
@@ -778,6 +782,20 @@ def validate_options
           },
           optional('agent') => Hash,
         }
+
+        unless ((username && api_key) || (application_credential_id && application_credential_secret)) ||
+          (username && api_key && application_credential_id && application_credential_secret)
```
[the remark this answers is not in the document]
Comment (fmoehler) [reply src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb:787]: We wanted to check that not all of these credentials are provided.

[the remark this answers is not in the document]
Comment (mvach) [reply src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb:787]: But then it should be 

```
unless ((username && api_key) || (application_credential_id && application_credential_secret)) &&
          !(username && api_key && application_credential_id && application_credential_secret)
```

[the remark this answers is not in the document]
Comment (a-hassanin) [reply src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb]: Agree, may be splitting this into two boolean variables make it more readable?
e.g.: using_api_key = username && api_key?

**Review by mvach — APPROVED** [review]
LGTM

**Review by a-hassanin — APPROVED** [review]
Looks good now!

**Pull request merged**